[TASK-7977] : feat migrate wallet context to separate hook

kushagrasarathe opened the pull request:
- migrated wallet context to separate hook
- managing wallet states using redux 
- also migrated zero dev context to a separate hook and redux for managing states
- optmized zerodev method calling to prevent un-necessary rpc calls if address already exists

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

Based on the comprehensive summary of changes, here are the release notes:

## Release Notes

- **Refactoring**
  - Restructured wallet and ZeroDev hooks from context-based to a dedicated hooks directory
  - Consolidated wallet and ZeroDev state management using Redux
  - Simplified context provider hierarchy

- **New Features**
  - Added Redux slices for wallet and ZeroDev state management
  - Introduced new custom hooks for wallet and ZeroDev functionality
  - Enhanced wallet connection and management capabilities

- **Improvements**
  - Improved modularity of wallet and authentication hooks
  - Streamlined import statements across multiple components
  - Updated state management approach for better performance and maintainability

- **Technical Updates**
  - Migrated wallet context to a more flexible hook-based implementation
  - Added new selector hooks for accessing wallet and ZeroDev state
  - Implemented comprehensive state types for wallet and ZeroDev features
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

kushagrasarathe committed:
feat: migrate walletContext to seprate hook + redux states

kushagrasarathe committed:
Merge branch 'peanut-wallet' into refactor/wallet-context

kushagrasarathe committed:
feat: fix useWallet and useZeroDev hooks

kushagrasarathe committed:
feat: update wallet-slice and update useWallet hooks

kushagrasarathe committed:
feat: refactor useZeroDev hook + use redux slice

kushagrasarathe committed:
fix: optimze zeroDev method calling to prevent unnecessary rpc calls

kushagrasarathe committed:
fix: remove commented code

kushagrasarathe commented on src/components/Global/Header/index.tsx:
```diff
@@ -8,7 +8,7 @@ import { HAMBURGER_LOTTIE, PEANUTMAN_LOGO } from '@/assets'
 import { useAppKit } from '@reown/appkit/react'
 
-import { useWallet } from '@/context/walletContext'
+import { useWallet } from '@/hooks/useWallet'
```
what is 14-14 here?

kushagrasarathe committed:
refactor: coderabbit comments

jjramirezn committed:
Merge branch 'peanut-wallet' into refactor/wallet-context

jjramirezn commented on src/hooks/useWallet.ts:
```diff
@@ -0,0 +1,213 @@
+        }
+    }, [selectedWallet, dispatch])
+
+    useEffect(() => {
```
This useEffect enters an infinite loop if trying to connect an already existing account (this does not happen currently in peanut.me)


https://github.com/user-attachments/assets/e0481457-eeda-427f-80e6-1aced04393bc

kushagrasarathe replied:
pushed a fix for this

jjramirezn committed:
style: fix

kushagrasarathe committed:
fix: infitine loop while adding account

- also fixed wallet connector images not rendering properly
- redux serialization errors

kushagrasarathe committed:
Merge branch 'peanut-wallet' into refactor/wallet-context

jjramirezn reviewed (approved):
Looks good, smoke test pass
